src/servant/tt_000/10E8.c

```diff
@@ -468,86 +463,80 @@ void func_801719E0(Entity* self) {
 }
 #endif
 
-#ifdef VERSION_PSP
-INCLUDE_ASM("servant/tt_000/nonmatchings/10E8", func_80171ED4);
-#else
 void func_80171ED4(s32 arg0) {
+    u16* dst;
+    u16* src;
     RECT rect;
     s32 i;
     s32 x;
-    u16* spriteBanks;
-    s16* src;
-    s16* dst;
+    SpriteParts** spriteBanks;
     Entity* e;
 
-    if (arg0 == 1 || arg0 == 3)
+    if ((arg0 == 1) || (arg0 == 3)) {
         ProcessEvent(NULL, true);
+        if (arg0 == 3) {
+            return;
+        }
+    }
 
-    if (arg0 == 3)
-        return;
-
-    dst = D_8006F3CC;
+    dst = g_Clut + 0x1400;
     src = D_80170448;
-    for (i = 0; i < 0x100; i++) {
+
+    for (i = 0; i < 256; i++) {
         *dst++ = *src++;
     }
 
-    dst = D_8006F42C;
-    spriteBanks = D_80170720;
-    src = spriteBanks;
-    for (i = 0; i < 0x20; i++) {
+    dst = g_Clut + 0x1430;
```

Comment:
Is &g_Clut[0xA18] matching?

Reply:
Yeah, `dst = &g_Clut[0x1430];` and similar are matching too. I'll be happier to change it if it makes it clearer!